plugins/thunderbolt/fu-plugin-thunderbolt.c

```diff
@@ -146,6 +147,89 @@ fu_plugin_thunderbolt_is_host (GUdevDevice *device)
 	return g_str_has_prefix (name, "domain");
 }
 
+static gchar *
+fu_plugin_thunderbolt_get_force_power_path (FuPlugin *plugin)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+
+	GList *devices;
+	const gchar *basepath;
+	g_autofree gchar *built_path = NULL;
+
+	devices = g_udev_client_query_by_subsystem (data->udev, "wmi");
+	for (GList *l = devices; l != NULL; l = l->next) {
+		GUdevDevice *device = l->data;
+		basepath = g_udev_device_get_sysfs_path (device);
```

Comment:
Do we need to check for `NULL` here?

Reply:
g_strcmp0 does that, it's the `0` bit at the end :)

Reply:
Not sure I understood your comment. If `g_build_path` gets `basepath` which is `NULL`, will it return an empty path or `"/"`?
If the latter, the `g_file_test` will succeed but for the wrong path.

Reply:
I think may as well add an extra guard to this to test for NULL.  The glib documentation isn't very clear about what happens in the case of a null string (only an empty one).